src/types.ts

```diff
@@ -75,35 +75,53 @@ type SectionType = 'autocomplete' | 'recommendations' | 'custom';
 
 export type SectionConfiguration = {
   type?: SectionType;
-  identifier: string;
   displayName?: string;
   numResults?: number;
   // This property will only take effect when using the component and not the hook
   displaySearchTermHighlights?: boolean;
   ref?: React.RefObject<HTMLElement>;
 };
 
-export interface AutocompleteSection extends SectionConfiguration {
+export interface AutocompleteSectionConfiguration extends SectionConfiguration {
   type?: 'autocomplete';
-  data: Item[];
+  indexSection: string;
+  /** @deprecated use indexSection field instead */
+  identifier?: string;
 }
 
-export interface RecommendationsSection extends SectionConfiguration {
+export interface RecommendationsSectionConfiguration extends SectionConfiguration {
   type: 'recommendations';
-  data: Item[];
+  indexSection: string;
+  podId: string;
   itemIds?: string[];
-  section?: string;
   term?: string;
+  /** @deprecated use podId field instead */
+  identifier?: string;
 }
 
-export interface CustomSection extends SectionConfiguration {
+export interface CustomSectionConfiguration extends SectionConfiguration {
   type: 'custom';
+  displayName: string;
+}
+
+export interface AutocompleteSection extends AutocompleteSectionConfiguration {
+  data: Item[];
+}
+
+export interface RecommendationsSection extends RecommendationsSectionConfiguration {
+  data: Item[];
+}
+
+export interface CustomSection extends CustomSectionConfiguration {
   data: Item[];
 }
 
```

Comment:
The above is becoming confusing, redundant and introduces unnecessary types that makes it hard to understand the different configs 😭 
<img width="973" alt="Screenshot 2023-12-26 at 10 58 14 PM" src="https://github.com/Constructor-io/constructorio-ui-autocomplete/assets/58053149/c65fc47e-8651-46b3-8400-52a96806939f">

What do you think of adding a utility type

- ` type WithData<T> = T & { data: Item[] };`
- `export type Section = WithData<AutocompleteSectionConfiguration> | WithData<RecommendationsSectionConfiguration> | WithData<CustomSectionConfiguration>;`

This will trim down types especially that they're not used out of this file

Reply:
I'm not against this change but the code you posted seems more confusing than what we have. It's shorter but it's more confusing. Let's talk about this today

Reply:
Leave the names alone but add comments